### PR TITLE
[SDKS-6478] add /admin/stats API

### PR DESCRIPTION
**Pull request by emmaz90**
# Split Evaluator

## What did you accomplish?
Added `/admin/stats` API to get environments and evaluator stats
calling stats should respond with evaluator uptime and health-check and a list of stats for each environment identified with the obfuscated authToken.

Environment stats should include:
- SplitCount
- SegmentCount
- Ready/Not Ready
- Last Evaluation
- Last Synchronization (telemetry, auth, events, impressions, segments, splits)
- Impressions mode

example:
```
{
  "uptime": "10d 2h 1m 4s",
  "healthcheck": {
    "version": "2.2.0",
    "sdk": "nodejs",
    "sdkVersion": "10.22.0"
  },
  "environments": {
    "######tkn1": {
      "splitCount": 20,
      "segmentCount": 5,
      "lastSynchronization": {
        "splits": 1674855033186,
        "segments": 1674855033286,
        "mySegments": 1674855033386,
        "impressions": 1674855033486,
        "impressionCount": 1674855033586,
        "events": 1674855033686,
        "telemetry": 1674855033786,
        "token": 1674855033886
      },
      "timeUntilReady": 833,
      "httpErrors": {},
      "ready": true,
      "impressionsMode": "OPTIMIZED"
    },
    "######tkn2": {
      "splitCount": 0,
      "segmentCount": 0,
      "lastSynchronization": {},
      "timeUntilReady": 0,
      "httpErrors": {
        "telemetry": {
          "401": 1
        }
      },
      "ready": false,
      "impressionsMode": "OPTIMIZED"
    }
  }
}
```

## How do we test the changes introduced in this PR?
Unit tests included

### CODE REVIEW

**Review by mmelograno — CHANGES_REQUESTED** [review]
Missing https://github.com/splitio/split-evaluator/blob/master/openapi/openapi.yaml changes with this new route.
In the comment of the PR can you share an example of the API call?

File: admin/admin.controller.js [hunk 1]
```diff
@@ -86,10 +89,48 @@ const uptime = (req, res) => {
...
+ *          - impressionsMode
+ *          - last Evaluation
+ *
+ * 200 - Everything is OK. Evaluator returns stats.
+ * 500 - Something is wrong or not ready yet. Not able to get stats.
+ * @param {*} req
+ * @param {*} res
+ */
+const stats = (req, res) => {
+  console.log('Running stats.');
```
Comment (mmelograno) [inline admin/admin.controller.js:110]: is this needed?

File: environmentManager/index.js [hunk 2]
```diff
@@ -109,6 +121,28 @@ const EnvironmentManagerFactory = (function(){
       return this.getFactory(authToken).manager();
     }
 
+    getTelemetry(authToken) {
+      if (!this.requireAuth) authToken = 'splitToken';
+      const environment = this.getEnvironment(authToken);
+      const telemetry = environment.telemetry;
+      const stats = {
+        splitCount: telemetry ? telemetry.splits.getSplitNames().length : undefined,
```
Comment (mmelograno) [inline environmentManager/index.js:129]: better 0 to undefined?

File: environmentManager/index.js [hunk 3]
```diff
@@ -109,6 +121,28 @@ const EnvironmentManagerFactory = (function(){
...
+        segmentCount: telemetry ? telemetry.segments.getRegisteredSegments().length : undefined,
```
Comment (mmelograno) [inline environmentManager/index.js:130]: same

File: environmentManager/index.js [hunk 4]
```diff
@@ -109,6 +121,28 @@ const EnvironmentManagerFactory = (function(){
...
+        lastSynchronization: telemetry ? telemetry.getLastSynchronization() : undefined,
```
Comment (mmelograno) [inline environmentManager/index.js:131]: same

File: environmentManager/index.js [hunk 5]
```diff
@@ -109,6 +121,28 @@ const EnvironmentManagerFactory = (function(){
...
+        timeUntilReady: telemetry ? telemetry.getTimeUntilReady(): undefined,
```
Comment (mmelograno) [inline environmentManager/index.js:132]: same

File: environmentManager/index.js [hunk 6]
```diff
@@ -109,6 +121,28 @@ const EnvironmentManagerFactory = (function(){
...
+        httpErrors: telemetry ? telemetry.httpErrors: undefined,
```
Comment (mmelograno) [inline environmentManager/index.js:133]: empty better

**Review by mmelograno — CHANGES_REQUESTED** [review]

File: admin/admin.controller.js (hunk 1 above)
Comment (emmaz90) [reply admin/admin.controller.js:110]: each endpoint on admin controller logs when is called, if you think that is not necessary I could remove it, I added it just to be consistent

Comment (mmelograno) [reply admin/admin.controller.js:110]: np

**Review by mmelograno — APPROVED** [review]

**Pull request merged**